### ISSUE TITLE
valid-jsdoc must not require @returns when @member block

### ISSUE THREAD

**Issue by gajus** [eslint/eslint#3744]
```js
/**
 * A promise that is resolved when window.onYouTubeIframeAPIReady is called.
 * The promise is resolved with a reference to global.YT object.
 *
 * @member {Object} iframeAPIReady
 */
iframeAPIReady = null;
```

```js
/**
 * A promise that is resolved when window.onYouTubeIframeAPIReady is called.
 * The promise is resolved with a reference to global.YT object.
 *
 * @var {Object} iframeAPIReady
 */
iframeAPIReady = null;
```

These produce an error:

```
Missing JSDoc @returns for function (valid-jsdoc)
```

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by gyandeeps** [comment]
Can you share the config for the rule?

**Comment by gajus** [comment]
```json
"valid-jsdoc": [
    1,
    {
        "requireParamDescription": false,
        "requireReturnDescription": false
    }
],
```

ESLint version v1.3.1.

**Comment by gyandeeps** [comment]
With that info I cant reproduce the error using the online demo. Also based on the code we only require return if the function has a `return` statement.

**Comment by gajus** [comment]
@gyandeeps Sorry. Try:

```js
/**
 * A promise that is resolved when window.onYouTubeIframeAPIReady is called.
 * The promise is resolved with a reference to global.YT object.
 *
 * @member {Object} iframeAPIReady
 */
iframeAPIReady = new Bluebird(function (resolve) {});
```

**Comment by gyandeeps** [comment]
I think regardless whether you have `@member` or not, this code should not throw a return error.
@eslint/eslint-team

**Comment by mysticatea** [comment]
If `new` is removed, this becomes the expected behavior.

I guess here https://github.com/eslint/eslint/blob/master/lib/util/source-code.js#L223 should includes `NewExpression`.

**Comment by nzakas** [comment]
Working on this.
